File: vscodejava/Hello/src/a0313/Ex01.java
package a0313;

public class Ex01 {
    public static void main(String[] args) {
        // 세자리 정수의 각 자릿수 총합을 출력하려 한다.
        // num = 374
        // 정수 374의 각 자릿수의 총합 : 14

        // 힌트1 3+7+4 =14
        // 힌트2 %:나머지, /:나누기를 적절히 이용
        // 힌트3 100으로 나누고 10으로 나누고 나머지 이용

        int a = 374;
        int a100 = a / 100;
        int a10 = a % 100 / 10;
        int a1 = a % 10;
        System.out.println(a100+a10+a1);

        int sum = a100+a10+a1;
        // 출력
        System.out.printf("정수 %d의 각 자릿수의 총합: %d",a,sum);
    }   
}

File: vsc/a0417/ex01.cpp
#include <stdio.h>

int main()
{
    int num;
    int sum;
    printf("양의 정수를 입력하세요");
    scanf("%d",&num);
    if(num < 0){
        printf("양의 정수를 입력해야합니다. \n");
        return 1;
    }

    while (num > 0)
    {
        sum = sum + (num % 10);  // 12345 % 10 = 5 일의 자리수 sum에 저장, 1234 % 10 = 4 일의 자리수 sum에 누적, 123 % 10 = 3 일의 자리수 sum에 누적 ...
        num = num / 10;  // 12345 / 10 = 1234 num에다가 저장, 1234 / 10 = 123 num에다가 저장, 123 / 10 = 12 num에다가 저장 ...
    }
    
    printf("%d",sum);
    return 0;
}
